swagger/smart_transfer_protocol_ws.yaml

```diff
@@ -0,0 +1,421 @@
+            - negotiatedUploads
+          properties:
+            datasetId:
+              type: string
+              format: did
+              description: Dataset identifier        
+            pushRequestId:
+              type: string
+              format: uuid
+              description: Request identifier of the started push flow              
+            negotiatedUploads:
+              type: array
+              description: Negotiated upload operations for requested bulk of files
+              items:
+                $ref: '#/components/schemas/NegotiatedUpload'
+
+  dataset/push/complete:
+    subscribe:
+      operationId: odf.dataset.push.complete
+      summary: Informs ODF-server the push operation is complete
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetPushComplete
+        name: DatasetPushComplete
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - pushRequestId
+          datasetId:
+            type: string
+            format: did
+            description: Dataset identifier       
+          pushRequestId:
+            type: string
+            format: uuid
+            description: Request identifier of the started push flow            
+
+  dataset/push/complete/confirmed:
+    publish:
+      operationId: odf.dataset.push.complete.confirmed
+      summary: Informs consumer the push completion is confirmed
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetPushCompleteConfirmed
+        name: DatasetPushCompleteConfirmed
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - pushRequestId
+            - newHead
+          datasetId:
+            type: string
+            format: did
+            description: Dataset identifier          
+          pushRequestId:
+            type: string
+            format: uuid
+            description: Request identifier of the started push flow
+          newHead:
+            type: string
+            format: multihash
+            description: New dataset head block hash after completion of this push request
+
+  dataset/error:
+    publish:
+      operationId: odf.dataset.error
+      summary: Informs about error happened during one of API operations with the dataset
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetError
+        name: DatasetError
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - errorDetails
+          properties:
+            datasetId:
+              type: string
+              format: did
+              description: Dataset identifier
+            errorDetails:
+              $ref: '#/components/schemas/ErrorDetails'
+
+components:
+  schemas:
+    TarOfBlocks:
```

Comment:
To make it a bit more extensible how about we call this a `ObjectsBatch` and it will look something like:
```yaml
objectType: MetadataBlock  # enum
encoding: application/tar  # allows us to add more encodings later
payload: ...
```
this would allow us later to potentially reuse this for in-band transfer of other objects if needed

Reply:
Good idea